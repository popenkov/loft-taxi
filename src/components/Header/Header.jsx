import { useActions } from '../../hooks/useActions';
import React, { useState } from 'react';
import cn from 'classnames';

import { ReactComponent as RoadLogo } from '../../assets/icons/road-logo.svg';
import { ReactComponent as TaxiLogo } from '../../assets/icons/taxi-logo.svg';
import { ReactComponent as BurgerIcon } from '../../assets/icons/burger-menu.svg';
import { ReactComponent as CloseIcon } from '../../assets/icons/close-menu.svg';
import styles from './header.module.scss';
import { Navigation } from './Navigation';
import ExitLink from './ExitLink/ExitLink';

export const Header = () => {
  const { logout } = useActions();
  const [isMenuOpen, setIsMenuOpen] = useState(false);

  const handleLogoutClick = () => {
    logout();
  };

  const handleBurgerMenuClick = () => {
    setIsMenuOpen(true);
  };

  const handleMenuCloseClick = () => {
    setIsMenuOpen(false);
  };

  return (
    <header className={styles.header}>
      <div
        className={cn(styles.mobileBurger, {
          [styles.open]: isMenuOpen,
        })}
        onClick={handleBurgerMenuClick}
        data-testid="open-btn"
      >
        <BurgerIcon />
      </div>
      <div
        className={cn(styles.closeButton, {
          [styles.open]: isMenuOpen,
        })}
        onClick={handleMenuCloseClick}
        data-testid="close-btn"
      >
        <CloseIcon />
      </div>
      <div
        className={cn(styles.headerContainer, {
          [styles.open]: isMenuOpen,
        })}
        data-testid="menu"
      >
        <div className={styles.logoContainer} data-testid="logo-container">
          <RoadLogo className={styles.roadLogo} />
          <TaxiLogo className={styles.taxiLogo} />
        </div>
        <div className={styles.navigationContainer}>
          <Navigation />

          <ExitLink handleClick={handleLogoutClick} />
        </div>
      </div>
    </header>
  );
};
