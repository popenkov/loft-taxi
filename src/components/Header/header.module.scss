@import '../../styles/variables.scss';
@import '../../styles/mixins.scss';
@import '../../styles/breakpoints.scss';

.header {
  width: 100%;
  height: 102px;
  background-color: $blackBg;

  &Container {
    width: 100%;
    max-width: 1440px;
    margin: 0 auto;
    padding: 20px 25px;
    @include flex-center-fullwidth;
  }
}

.logoContainer {
  @include flex-center-vert;
}

.roadLogo {
  @include box(61px);
  margin-right: 16px;
}

.navigationContainer {
  @include flex-center-vert;
}

.logoutContainer {
  margin-left: 16px;
}
