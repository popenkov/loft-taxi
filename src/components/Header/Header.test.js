import { screen, fireEvent } from '@testing-library/react';
import { Header } from './Header';
import { renderTestApp } from '../../test/helpers/renderTestApp';

describe('header component', () => {
  let container;

  beforeEach(() => {
    container = renderTestApp(<Header />, {
      route: '/',
      initialState: {},
    });
  });

  test('renders correctly', async () => {
    expect(<Header />).toMatchSnapshot();
  });

  test('shows "Профиль"', () => {
    expect(container.getByText('Профиль')).not.toBeNull();
  });

  test('shows logo', () => {
    screen.debug();
    const logoContainer = container.getByTestId('logo-container');
    expect(logoContainer).toBeVisible();
  });
});
