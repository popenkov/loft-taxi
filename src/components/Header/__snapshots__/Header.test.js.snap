// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header component renders correctly 1`] = `<Header />`;
